hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootControllerTest.java

```diff
@@ -362,13 +370,77 @@ public void tryToFinishAnUpdateProcessAfterItHasBeenFinished() throws Exception
                 .andDo(MockMvcResultPrinter.print()).andExpect(status().isGone());
     }
 
+    @Test
+    @Description("Controller sends attribute update request after device successfully closed software update.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 1),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 2),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 2),
+            @Expect(type = ActionCreatedEvent.class, count = 2), @Expect(type = ActionUpdatedEvent.class, count = 2),
+            @Expect(type = TargetUpdatedEvent.class, count = 6), @Expect(type = TargetPollEvent.class, count = 4),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 6),
+            @Expect(type = TargetAttributesRequestedEvent.class, count = 1) })
+    public void testAttributeUpdateRequestSendingAfterSuccessfulDeployment() throws Exception {
+        final String targetControllerId = "922";
+        final DistributionSet ds1 = testdataFactory.createDistributionSet("1");
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("2");
```

Comment:
you don't need second distribution set, you can reassign the first one (leaving prefix blank during creation). Don't forget to update the action and tune the number of expected events.